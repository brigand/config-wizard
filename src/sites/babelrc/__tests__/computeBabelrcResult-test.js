import f from '../computeBabelrcResult';

it('edge: bleeding', () => {
  const res = f({edge: 'bleeding', support: {browsers: ['> 1%']}});
  expect(res).toEqual({
    deps: ['babel-preset-env', 'babel-preset-stage-0'],
    rc: {
      presets: [
        ['env', {
          targets: {
            browsers: ['> 1%'],
          }
        }],
        'stage-0',
      ],
    },
  })
});

it('edge: safe, framework: nodeApp', () => {
  const res = f({edge: 'safe', framework: 'nodeApp', support: {node: 6}});
  expect(res).toEqual({
    deps: ['babel-preset-env'],
    rc: {
      presets: [
        ['env', {
          targets: {
            node: 6,
          }
        }],
      ],
    },
  })
});

const snapshots = [
  {edge: 'cutting', framework: 'react', support: {browsers: ['> 1%']}},
  {edge: 'cutting', framework: 'angular1', support: {browsers: ['> 1%']}},
  {edge: 'cutting', framework: 'angular2', support: {browsers: ['> 1%']}},
  {edge: 'cutting', framework: 'nodePackage', support: {browsers: [], node: null}},
  {edge: 'cutting', framework: 'none', support: {browsers: ['> 1%'], node: null}},
];

snapshots.forEach((opts) => {
  it(`edge: ${opts.edge}, framework: ${opts.framework}, support: ${JSON.stringify(opts.support)}`, () => {
    const res =f(opts);
    expect(res).toMatchSnapshot();
  });
});
