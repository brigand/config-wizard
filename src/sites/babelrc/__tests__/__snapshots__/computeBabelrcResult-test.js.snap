exports[`test edge: cutting, framework: angular1, support: {"browsers":["> 1%"]} 1`] = `
Object {
  "deps": Array [
    "babel-preset-env",
    "babel-plugin-transform-class-properties",
    "babel-plugin-angularjs-annotate@^0.7.0",
  ],
  "rc": Object {
    "plugins": Array [
      "transform-class-properties",
      "angularjs-annotate",
    ],
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "> 1%",
            ],
          },
        },
      ],
    ],
  },
}
`;

exports[`test edge: cutting, framework: angular2, support: {"browsers":["> 1%"]} 1`] = `
Object {
  "deps": Array [
    "babel-preset-env",
    "babel-plugin-transform-decorators",
  ],
  "rc": Object {
    "plugins": Array [
      "transform-decorators",
    ],
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "> 1%",
            ],
          },
        },
      ],
    ],
  },
}
`;

exports[`test edge: cutting, framework: nodePackage, support: {"browsers":[],"node":null} 1`] = `
Object {
  "deps": Array [
    "babel-preset-latest",
  ],
  "rc": Object {
    "presets": Array [
      "latest",
    ],
  },
}
`;

exports[`test edge: cutting, framework: none, support: {"browsers":["> 1%"],"node":null} 1`] = `
Object {
  "deps": Array [
    "babel-preset-env",
  ],
  "rc": Object {
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "> 1%",
            ],
          },
        },
      ],
    ],
  },
}
`;

exports[`test edge: cutting, framework: react, support: {"browsers":["> 1%"]} 1`] = `
Object {
  "deps": Array [
    "babel-preset-env",
    "babel-preset-react",
    "babel-plugin-transform-class-properties",
    "babel-plugin-transform-object-rest-spread",
  ],
  "rc": Object {
    "plugins": Array [
      "transform-class-properties",
      "transform-object-rest-spread",
    ],
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": Array [
              "> 1%",
            ],
          },
        },
      ],
      "react",
    ],
  },
}
`;
